scripts/typescript/src/bcdb/bcdb.ts

```diff
@@ -0,0 +1,352 @@
+    });
+    log(`OK`);
+
+    //-- Load the sheets
+
+    const project = {
+      name: "BCDB",
+      model_version: "1.0.0",
+      node: ["Project"],
+      collection: new Array<ICollection>(),
+      material: new Array<IMaterial>(),
+    } satisfies IProject;
+
+    // Store references as: doi => IReference
+    const references = new Map<string, IReference>();
+    // Store blocks as: bigsmiles => {...}
+    const blocks = new Map<string, { index: number; polymer: string; bigsmiles: string; name: string }>();
+
+    //-- Loads polymer/name/bigsmiles table
+    {
+      const sheetName = options.sheets[0];
+      process.stdout.write(`Loading sheet ${sheetName} ...\n`);
+      const rows: { [key: string]: string }[] = XLSX.utils.sheet_to_json(workBook.Sheets[sheetName], {
+        header: "A",
+      });
+
+      log(`-- Found ${rows.length} row(s)`);
+
+      rows
+        .slice(1) // Skip header row.
+        .forEach((row, index) => {
+          const polymer = row["A"];
+          const name = row["B"];
+          const bigsmiles = row["C"];
+          const entry = { index, polymer, name, bigsmiles };
+          blocks.set(bigsmiles, entry);
+        });
+      log(`Found ${JSON.stringify([...blocks.values()], null, " ")} (${blocks.size} rows)`);
+      log(`Storing blocks meta information ... DONE`);
+    }
+
+    {
+      const sheetName = options.sheets[1];
+      process.stdout.write(`Loading sheet ${sheetName} ...\n`);
+      const rows: { [key: string]: string }[] = XLSX.utils.sheet_to_json(workBook.Sheets[sheetName], {
+        header: "A",
+      });
+      log(`-- Found ${rows.length} row(s)`);
+      log(`-- Converting polymer rows to JSON ...`);
+
+      rows
+        .slice(2) // Skip 2 header rows
+        .forEach((row, index, arr) => {
+          index % 100 == 0 && process.stdout.write(`-- Loading row ${index}/${arr.length - 1}\n`);
+
+          //-- reference
+          //   to store DOI and ORCIDs
+          const doi = row[Column.DOI];
+          const author = row[Column.ORCID];
+          let reference: IReference | undefined = references.get(doi);
+          if (!reference) {
+            reference = {
+              node: ["Reference"],
+              doi: doi,
+              author: [author], // TODO: parse array
+              type: "", // TODO: pick a reference-type key from vocab
+            };
+            if (reference.doi) {
+              references.set(reference.doi, reference);
+            } else {
+              error(index, `Unable to store this reference, it has no doi (we use it as a key)`);
+            }
+          }
+          //log(`Reference read: ${reference.doi}`);
+
+          //-- citation
+          //   to reference the reference (haha)
+          const citation: ICitation = {
+            // uid: will be determined by CriptJSON serializer
+            name: `RCBC Citation ${index}`,
+            node: ["Citation"],
+            type: "reference",
```

Comment:
Same remark, I picked `reference` but I clearly don't know if it is the correct value.

> "extracted_by_human",
                    "extracted_by_nlp",
                    "extracted_by_algorithm", // could be this too
                    "derived_from",
                    "replicated",
                    "reference"

Reply:
extracted_by_human